syntaxes/Asciidoctor.json

```
@@ -841,11 +841,14 @@
       "patterns": [
         {
           "name": "markup.other.footnote.asciidoc",
-          "begin": "(?<!\\\\)(footnote(?:ref)?):\\[",
+          "begin": "(?<!\\\\)(footnote(?:ref)?):(.*?)\\[",
```

I think we should use:


```suggestion
          "begin": "(?<!\\\\)footnote(?:(ref):|:([\w-]+)?)\[(?:|(#.*?[^\\]))\]",
```